container/wait_test.go

```diff
@@ -12,14 +12,18 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// mockWaitForStatus mocks waitForStatus() to fake wantedStatus to happen for namespace.
 func mockWaitForStatus(t *testing.T, m *mocks.CommonAPIClient, namespace string, wantedStatus StatusType) {
 	m.On("TaskList", mock.Anything, types.TaskListOptions{
 		Filters: filters.NewArgs(filters.KeyValuePair{
 			Key:   "label",
 			Value: "com.docker.stack.namespace=" + namespace,
 		}),
-	}).Once().
+	}).
+		Once().
 		Return([]swarm.Task{}, nil)
+
+	// status:
```

Comment:
what do you mean?

Reply:
It means mocking the status. We have 10s of mock functions like this and it's obvious that we're creating mocks inside the tests. These comments meant to separate different kind of mocks from each other so we can easily understand what features we're mocking with what group of mock calls.

What kind of comments do you prefer?

Reply:
Yes, but the function tells the story, I mean this comment is pointless here and could be removed.

Reply:
`status: ` is useless. The function name is already `mockWaitForStatus`, the called function is `mockStatus`. It's self explanatory already.

Reply:
Functions **ONLY** tells the story if you have time to read **each of the mocks** and check the implementation details so you can understand why and how they are used. It's hard to understand mocks if you don't follow this flow.

In programming, comments are meant to be used to make code more understandable. This is especially important for mocks because they can be very hard to understand.

See https://en.wikipedia.org/wiki/Comment_(computer_programming)

Reply:
Yes,, comments should give more explanation. If a comment is the name of the following function, then it's worthless.